emgapi/metagenomics_exchange.py

```diff
@@ -108,14 +108,20 @@ def add_analysis(self, mgya: str, sequence_accession: str):
         data = self.generate_metadata(mgya, sequence_accession)
         try:
             response = self.post_request(endpoint="datasets", data=data)
+            response.raise_for_status()  # Ensure we raise for HTTP errors
+            return response
         except HTTPError as http_error:
             try:
                 response_json = http_error.response.json()
                 logging.error(f"API response content: {response_json}")
-            except:
-                pass
-            raise http_error
-        return response
+            except ValueError:  # Catch JSON decoding errors
```

Comment:
Wouldn't this raise an `requests.exceptions.JSONDecodeError` rather than a `ValueError` 🤔 ?

Reply:
you are right, one is a subclass of another. Fixed!